tests/e2e/common/retry_test_mixins.py

```diff
@@ -323,3 +330,90 @@ def test_retry_abort_close_operation_on_404(self):
                 self.assertTrue(
                     expected_message_was_found, "Did not find expected log messages"
                 )
+
+    # I really want to use pytest.mark.parametrize here but it doesn't work for tests defined in a
+    # unittest.TestCase. Our test suite needs some reorganisation anyway, since right now all of the
+    # tests are blended into the PySQLCoreTestSuite. I'll leave this as-is for now.
+
+    # We only test up to 5 retries because this mixin's stop_after_attempts_count is 5.
+    def _base_retry_max_redirect_raises(self, max_redirects, expected_call_count):
+        """GIVEN the connector is configured with a custom max_redirects
+        WHEN the DatabricksRetryPolicy is created
+        THEN the connector raises a MaxRedirectsError if that number is exceeded
+        """
+        # Code 302 is a redirect
+        with mocked_server_response(
+            status=302, redirect_location="/foo.bar/"
+        ) as mock_obj:
+            with self.assertRaises(MaxRetryError) as cm:
+                with self.connection(
+                    extra_params={
+                        **self._retry_policy,
+                        "_retry_max_redirects": max_redirects,
+                    }
+                ) as conn:
+                    pass
+            assert "too many redirects" == str(cm.exception.reason)
+            # Total call count should be 3 (original + 2 retries)
+            assert mock_obj.return_value.getresponse.call_count == expected_call_count
+
+    def test_retry_max_redirects_raises_1_2(self):
```

Comment:
Are all of these actually testing different behavior modes?  If not, recommend removing the ones that are functional duplicates.

Reply:
It was mainly due diligence to confirm that the exception raised as expected for multiple different configs. Later, I added the assertion that the exception text is "too many redirects" rather than "too many $statusCode responses" which accomplishes the same thing. Will remove these faux-paramterised tests.

Reply:
Fixed in 21d418d34dc74fcbc6ffe19fecb97215c7dea07a